amida-tech/recruitment-registry
# Feature/rr 852

If this PR fixes a bug, you _must_ add test cases representative of the bug.

Please refer to [Tettra](https://app.tettra.co/teams/amida/pages/amida-pull-request-and-code-review-guide) for PR review guidelines.

#### What does this PR do? Implements patch when patch is a full survey objects

#### Related JIRA tickets: RR 852

#### How should this be manually tested?

1) Post a survey.
2) Get the posted survey from server.
3) Change various fields to create a patch but leave all fields that need to be the same as is. Also add a 'complete: true' to top level. That is for backward compatibility.
4) Remove survey 'id' from patch and call surveys/{id} [PATCH] with the patch object.
5) Check if the fields are changed as expected.

#### Background/Context

#### Screenshots (if appropriate):
